lib/strava/models/mixins/start_date_local.rb

```diff
@@ -0,0 +1,29 @@
+module Strava
+  module Models
+    module Mixins
+      module StartDateLocal
+        extend ActiveSupport::Concern
+
+        included do
+          property 'start_date_local', transform_with: ->(v) { ::Time.parse(v) }
+          def start_date_local
+            dt = self['start_date_local']
+
+            # some strava object do not contain a timezone property i.e. `Lap`
+            unless try(:timeout)
+              shift = "-0#{((start_date - dt) / 3600).to_i}:00"
+              return ::Time.new(dt.year, dt.month, dt.day, dt.hour, dt.min, dt.sec, shift)
+            end
```

Comment:
You are still relying on invoking a method that doesn't exist, then relying on an exception to do something else.

Plus you're invoking the wrong method, so you have lack of test coverage here :) The property is `timezone`, not `timeout`.

You want something like this:

```
if key?(:timezone) # possibly another way to check, but not try calling a method

elsif timezone.include? ...
elsif timezone.include? ...
elsif
   # raise an error?
end
```

Make sure to have tests for every branch.